pkg/model/crd.go

```diff
@@ -657,6 +657,45 @@ func (r *CRD) GetAllRenames(op OpType) (map[string]string, error) {
 	return renames, nil
 }
 
+// FindIdentifierInShape returns true and the resource's identifier iff shape
+// contains a matching field or corresponding plural meaning it can be populated
+// using the resource identifier.
+func (r *CRD) FindIdentifierInShape(
```

Comment:
refactoring this into a common util file instead of crd.go

Reply:
Nice! Maybe we could also reuse it in here https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/code/set_resource.go#L855 :)